cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -441,13 +449,18 @@ func (ks *kube2sky) updateService(oldObj, newObj interface{}) {
 	ks.newService(newObj)
 }
 
-func newEtcdClient(etcdServer string) (*etcd.Client, error) {
+func newEtcdClient(config *etcdConfig) (*etcd.Client, error) {
 	var (
 		client *etcd.Client
 		err    error
 	)
+	etcdServer := config.ServerList[0]
+	httpClient, err := config.newHTTPClient()
+	if err != nil {
+		return nil, fmt.Errorf("unable to create HTTP client: %v", err)
+	}
 	for attempt := 1; attempt <= maxConnectAttempts; attempt++ {
```

Comment:
It seems that the second loop is not similarly ensuring connectivity as the first does - creating an etcd client does not verify that any of the endpoints is reachable, hence the version query in the first loop.
I can try hiding the same check in `newClient` to achieve the same effect with just a single loop though.